crates/uv/tests/it/tool_run.rs

```diff
@@ -1889,6 +1889,73 @@ fn tool_run_python_at_version() {
     Audited in [TIME]
     "###);
 
+    // The @ is optional.
+    uv_snapshot!(context.filters(), context.tool_run()
+        .arg("python3.11")
+        .arg("--version"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    Python 3.11.[X]
+
+    ----- stderr -----
+    Resolved in [TIME]
+    "###);
+
+    // Dotless syntax also works.
+    uv_snapshot!(context.filters(), context.tool_run()
+        .arg("python311")
+        .arg("--version"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    Python 3.11.[X]
+
+    ----- stderr -----
+    Resolved in [TIME]
+    "###);
+
+    // Other implementations like PyPy also work. PyPy isn't currently in the test suite, so
+    // specify CPython and rely on the fact that they go through the same codepath.
+    uv_snapshot!(context.filters(), context.tool_run()
+        .arg("cpython311")
+        .arg("--version"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    Python 3.11.[X]
+
+    ----- stderr -----
+    Resolved in [TIME]
+    "###);
+
+    // But short names don't work in the executable position (as opposed to with -p/--python). We
+    // interpret those as package names.
+    uv_snapshot!(context.filters(), context.tool_run()
+        .arg("cp311")
+        .arg("--version"), @r"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+
+    ----- stderr -----
+      × No solution found when resolving tool dependencies:
+      ╰─▶ Because cp311 was not found in the package registry and you require cp311, we can conclude that your requirements are unsatisfiable.
```

Comment:
Does this snapshot break when someone publishes `cp311` to PyPI?

Reply:
Yes, but we should squat or advocate for those names to be banned.

Reply:
It's just as bad, really. We try to make them reproducible wherever possible. However, I think this specific case is fine.